src/test/java/mate/academy/service/UserServiceTest.java

```diff
@@ -0,0 +1,105 @@
+package mate.academy.service;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.impl.UserServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class UserServiceTest {
+    private static final String EMAIL = "tom@gmail.com";
+    private static final String PASSWORD = "1234";
+    private static final Long WRONG_ID = 999L;
+    private static final String WRONG_EMAIL = "invalid_email";
+    private UserDao userDao;
+    private UserService userService;
+    private PasswordEncoder passwordEncoder;
+    private User actualUser;
+    private Role userRole;
+
+    @BeforeEach
+    void setUp() {
+        userDao = Mockito.mock(UserDao.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        userService = new UserServiceImpl(userDao, passwordEncoder);
+        userRole = new Role(Role.RoleName.USER);
+        actualUser = getActualUser();
+    }
+
+    @Test
+    void save_Ok() {
+        userDao.save(actualUser);
```

Comment:
You should not call userDao.save(actualUser) directly in the test method. Instead, call the method from the userService under test and then verify the userDao's save method was called with the correct arguments using Mockito.verify().